BaseClasses.py

```diff
@@ -571,10 +570,10 @@ def sweep_for_events(self, key_only=False, locations=None):
 
 
     def can_reach_blue(self, region, player):
-        return region in self.reachable_regions[player] and self.reachable_regions[player][region] in [CrystalBarrier.Blue, CrystalBarrier.Either]
+        return region in self.reachable_regions[player] and self.reachable_regions[player][region] in [CrystalBarrier.Blue, CrystalBarrier.Either] and self.can_hit_crystal(player)
 
     def can_reach_orange(self, region, player):
-        return region in self.reachable_regions[player] and self.reachable_regions[player][region] in [CrystalBarrier.Orange, CrystalBarrier.Either]
+        return region in self.reachable_regions[player] and self.reachable_regions[player][region] in [CrystalBarrier.Orange, CrystalBarrier.Either] and self.can_hit_crystal(player)
```

Comment:
You don't need to hit a switch to be in orange state. :)

Reply:
Sometimes you do though!
I.e. you have to be in blue state to go to region X and only then reach orange state to go to region Y.